src/panels/config/integrations/integration-panels/zwave_js/dialog-zwave_js-remove-failed-node.ts

```diff
@@ -0,0 +1,240 @@
+                  ${this.hass.localize(
+                    "ui.panel.config.zwave_js.remove_failed_node.introduction"
+                  )}
+                </div>
+              </div>
+              <mwc-button slot="primaryAction" @click=${this._startExclusion}>
+                ${this.hass.localize(
+                  "ui.panel.config.zwave_js.remove_failed_node.remove_device"
+                )}
+              </mwc-button>
+            `
+          : ``}
+        ${this._status === "started"
+          ? html`
+              <div class="flex-container">
+                <ha-circular-progress active></ha-circular-progress>
+                <div class="status">
+                  <p>
+                    <b>
+                      ${this.hass.localize(
+                        "ui.panel.config.zwave_js.remove_failed_node.in_progress"
+                      )}
+                    </b>
+                  </p>
+                </div>
+              </div>
+            `
+          : ``}
+        ${this._status === "failed"
+          ? html`
+              <div class="flex-container">
+                <ha-svg-icon
+                  .path=${mdiCloseCircle}
+                  class="error"
+                ></ha-svg-icon>
+                <div class="status">
+                  <p>
+                    ${this.hass.localize(
+                      "ui.panel.config.zwave_js.remove_failed_node.removal_failed"
+                    )}
+                  </p>
+                  ${this._error
+                    ? html` <p><em> ${this._error.message} </em></p> `
+                    : ``}
+                </div>
+              </div>
+              <mwc-button slot="primaryAction" @click=${this.closeDialog}>
+                ${this.hass.localize("ui.panel.config.zwave_js.common.close")}
+              </mwc-button>
+            `
+          : ``}
+        ${this._status === "finished"
+          ? html`
+              <div class="flex-container">
+                <ha-svg-icon
+                  .path=${mdiCheckCircle}
+                  class="success"
+                ></ha-svg-icon>
+                <div class="status">
+                  <p>
+                    ${this.hass.localize(
+                      "ui.panel.config.zwave_js.remove_failed_node.removal_finished",
+                      "id",
+                      this._node!.node_id
+                    )}
+                  </p>
+                </div>
+              </div>
+              <mwc-button
+                slot="primaryAction"
+                @click=${this.closeDialogFinished}
+              >
+                ${this.hass.localize("ui.panel.config.zwave_js.common.close")}
+              </mwc-button>
+            `
+          : ``}
+      </ha-dialog>
+    `;
+  }
+
+  private async _startExclusion(): Promise<void> {
+    if (!this.hass) {
+      return;
+    }
+    this._status = "started";
+    try {
+      this._subscribed = await this.hass.connection.subscribeMessage(
```

Comment:
This could be a potential memory leak, if the dialog is now closed (disconnected) while we are awaiting this, we will never be unsubscribed.

Reply:
just to make sure I'm following correctly:
1. We start awaiting subscribeMessage
2. The dialog is closed while subscribeMessage is still being awaited
3. We try to unsubscribe (which doesn't do anything because we aren't subscribed yet)
4. The subscribeMessage call finishes so we're subscribed in the background, but the application thinks we've already unsubscribed so we're stuck with the subscription in the background

That an accurate assessment?

Reply:
Yep! :-)